docs/extensions/example.md

```diff
@@ -0,0 +1,233 @@
+]
+```
+
+```ts
+// Group all versions for the same dependency together.
+const groupedDeps = groupBy(deps, dep => dep.name);
+```
+
+Since our package list contains multiple instances of the same dependency, we
+need to group each instance together to find duplicate versions. Deno's
+convenient `groupBy` function does this for us automatically and we just need to
+tell it which field to group by using `dep => dep.name`.
+
+This will transform our package list into the following:
+
+```
+{
+  accepts: [
+      { name: "accepts", version: "1.3.8", type: "npm" },
+      { name: "accepts", version: "1.0.0", type: "npm" }
+  ],
+  "array-flatten": [ { name: "array-flatten", version: "1.1.1", type: "npm" } ]
+}
+```
+
+```ts
+// Reduce each dependency to a list of its versions.
+const reducedDeps = mapValues(groupedDeps, deps => deps.map(dep => dep.version));
+```
+
+Since our dependency structure now contains useless information like `name` and
+`type`, we map each of these grouped values to contain only the version numbers
+for each dependency.
+
+This results in a simple array with all dependencies and their versions:
+
+```
+{
+  accepts: ["1.3.8", "1.0.0"],
+  "array-flatten": ["1.1.1"]
+}
+```
+
+```ts
+for (const [dep, versions] of Object.entries(reducedDeps)) {
+```
+
+Since we now have an object containing all dependencies and the required
+versions, we can iterate over all fields in this object to check the number of
+versions it has.
+
+```ts
+// Deduplicate identical versions.
+const distinctVersions = distinct(versions);
+```
+
+But before we can check the versions themselves, we need to make sure all the
+versions are actually unique. Some lockfiles might specify the same version
+multiple times, so we need to ensure we filter duplicate versions.
+
+```ts
+// Print all dependencies with more than one version.
+if (distinctVersions.length > 1) {
+    console.log(`${dep}: `, distinctVersions);
+}
+```
+
+With all versions deduplicated, we can finally print out each dependency with
+more than one version in our lockfile.
+
+For our example, the output looks like this:
+
+```
+accepts: [ "1.3.8", "1.0.0" ]
+```
+
+And that's all the code we need to check for duplicates. Now we only need to
+install it and we can use it for any lockfile we encounter in the future:
+
+```
+phylum extension install ./duplicates
+phylum duplicates ./package-lock.json
+```
+
+Currently none of our code does any external I/O, we're completely contained
+within the Deno sandbox. Let's change that. Instead of the `console.log`, we'll
+now write our output to a file instead:
+
+```ts
+await Deno.writeTextFile(
+    "./duplicates.txt",
+    `${dep}: ${distinctVersions}\n`,
+    { append: true }
+);
+```
+
+When replacing the `console.log` with this function call and executing our
+extension, you'll run into the following error:
+
+TODO: This error likely needs to be updated on release.
+
+```
+Execution failed: Error: Requires write access to "./duplicates.txt"
+```
```

Comment:
I have no idea yet how we should document this since I don't know what error we'll end up with. I would recommend opening an issue and then updating it once the error message is fixed?

Reply:
Agreed. In the meantime we may want to add a note that mentions that that error is coming straight from Deno and/or is temporary and will be changed, just in case upstream doesn't get the changes we need in time.